File: 1925-count-square-sum-triples/1925-count-square-sum-triples.java
class Solution {
    int solve(int i, int n) {
        if (i >= n)
            return 0;
        return inner(i, i + 1, n) + solve(i + 1, n);
    }

    int inner(int i, int j, int n) {
        if (j >= n)
            return 0;
        int t = i * i + j * j;
        int k = (int) Math.sqrt(t);
        int c = (k * k == t && k <= n) ? 2 : 0;
        return c + inner(i, j + 1, n);
    }

    public int countTriples(int n) {
        return solve(1, n);
    }
}

File: 1925-count-square-sum-triples/1925-count-square-sum-triples.cpp
class Solution {
public:
    int countTriples(int n) {
        int cnt  = 0;
        for(int i = 1; i <= n ; i++){
            for(int j =i+1; j <= n; j++){
                for(int k = j+1; k <= n; k++){
                   long long a = i * i;
                   long long b = j * j;
                   long long c = k * k;
                   if( a + b == c || a + c == b || c + b == a){
                        cnt += 2;
                   }
                }
            }
        }
    return cnt;
    }
};